ns/list.n3

```diff
@@ -0,0 +1,98 @@
+@prefix vs: <http://www.w3.org/2003/06/sw-vocab-status/ns#> .
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
+
+@prefix list: <http://www.w3.org/2000/10/swap/list#>.
+@prefix : <#>.
+
+list:append a rdf:Property, owl:FunctionalProperty;
+    rdfs:label "append";
+    rdfs:domain log:List;
+    rdfs:comment """
+Iff the subject is a list of lists and the concatenation of all those lists 
+is the object, then this is true.
+eg  ( (1 2) (3 4) ) list:append (1 2 3 4).
+The object can be calculated as a function of the subject.
+""";
+    rdfs:isDefinedBy <>.
+
+list:first a rdf:Property, owl:FunctionalProperty;
+    rdfs:label "first";
+    rdfs:domain log:List;
+    rdfs:comment """
+Iff the subject is a list and the object is the first thing that list, then this is true.
+The object can be calculated as a function of the list.
+""";
+    vs:term_status "unstable";
+    rdfs:isDefinedBy <>.
+
+list:in a rdf:Property;
+    rdfs:label "in";
+    rdfs:range log:List;
+    rdfs:comment """
+Iff the object is a list and the subject is in that list, then this is true.
+""";
+    rdfs:isDefinedBy <>.
+
+list:last a rdf:Property, owl:FunctionalProperty;
+    rdfs:label "in";
+    rdfs:domain log:List;
+    rdfs:comment """
+Iff the subject is a list and the object is the last thing that list, then this is true.
+The object can be calculated as a function of the list.
+""";
+    rdfs:isDefinedBy <>.
+
+list:length a rdf:Property, owl:FunctionalProperty;
+    rdfs:label "length";
+    rdfs:domain log:List;
+    rdfs:range xsd:integer;
+    rdfs:comment """
+Iff the subject is a list and the object is the last thing that list, then this is true.
+The object can be calculated as a function of the list.
+""";
+    vs:term_status "unstable";
+    rdfs:isDefinedBy <>.
+
+list:member a rdf:Property;
+    rdfs:label "member";
+    rdfs:domain log:List;
+    rdfs:comment """
+Iff the subject is a list and the object is in that list, then this is true.
+""";
+    rdfs:isDefinedBy <>.
+
+list:memberAt a rdf:Property;
+    rdfs:label "memberAt";
+    rdfs:domain log:List;
+    rdfs:comment """
+Iff the subject is a list composed of a list and integer position,
+and the object exists within that list at the specified position, then this is true.
+The object can be calculated as a function of the subject.
+The position can be calculated as a function of the first component of the subject and the object.
+""";
+    vs:term_status "unstable";
+    rdfs:isDefinedBy <>.
+
+list:remove a rdf:Property, owl:FunctionalProperty;
+    rdfs:label "remove";
+    rdfs:domain log:List;
+    rdfs:range log:List;
+    rdfs:comment """
+Iff the subject is a list of two lists and the object is a list composed of the
+members of the first component of the subject up to but not including the second component
+of the subject, then this is true.
+eg ( (1 2 3 4) (3 4) ) list:remove (1 2).
+The object can be calculated as a function of the subject.
+""";
```

Comment:
This was my interpretation of the comment on how this would work, but I'm not sure it's too useful.

Reply:
Certainly the following could be useful (?)
```
( (1 2 3 4) ?l) list:remove (1 2)
```

Reply:
This could be achieved by
```
( (1 2) ?l) list:append (1 2 3 4)
```

Reply:
As you may have inferred from my comment, it didn't actually occur to me that these are new builtins! Sorry - I thought that they were still from the "old" namespace document - since they are still following the previous (and incomplete) way of describing builtins. It's unclear why you're diverging from the more structured builtin description, which we discussed in detail during the meeting.

Reply:
I was re-purposing the builtin comments already in the namespace documents. Can you provide a reference to the more detailed descriptions? Do they exist for these builtins? I didn't find the information in [Community wish-list of N3 built-ins](https://docs.google.com/document/d/1ByEWSADIvebHRrBda0HhdKhHhp444mYM7BAapXVIJoM/edit?pli=1) too informative, but perhaps there's a more definitive document I'm missing. IMO, we should either get these into the repo, or provide a definitive list of working documents with references in the README.

Reply:
Maybe we change the description to something like the following, to allow a more extended use case:

> Iff the subject is a list of two lists, the second component is a subset of the first component and the object is a list composed of the members of the first component of the subject with all members of the second component of the subject removed, matching left to right.
> eg `( (1 2 3 4) (2 3) ) list:remove (1 4)`.
> The object can be calculated as a function of the subject.

Reply:
The idea is to have new builtin descriptions that are sufficiently detailed (incl. domains, flow patterns, etc) to allow for consistent implementations - hence our efforts towards [more structured builtin descriptions](https://docs.google.com/document/d/19LSGC7OgGK8oGdPTTrtkgQzYTU5Gq5R-v5FTHf2sHEQ/edit).